### PR TITLE
Support for diffrent mouse buttons in UI buttons

### PR DESCRIPTION
**Pull request by TheRawMeatball**

### CODE REVIEW

**Comment by CleanCut** [comment]
It panics when I attempt to run the `ui` example on this branch with `cargo run --example ui`.  See attached backtrace.

[backtrace.txt](https://github.com/bevyengine/bevy/files/5147105/backtrace.txt)

**Comment by TheRawMeatball** [comment]
The example runs fine on my machine, and the backtrace suggests a missing font. Are you sure all assets are where they need to be?

**Comment by CleanCut** [comment]
Hmm...the same command works fine now. Dunno what was wrong before, as I didn't do anything but checkout your branch of `bevy`.  I withdraw my crash comment. 🤷‍♂️

File: crates/bevy_input/src/input.rs [hunk 1]
```diff
@@ -3,9 +3,9 @@ use std::hash::Hash;
 
 /// A "press-able" input of type `T`
 pub struct Input<T> {
-    pressed: HashSet<T>,
-    just_pressed: HashSet<T>,
-    just_released: HashSet<T>,
+    pub pressed: HashSet<T>,
+    pub just_pressed: HashSet<T>,
+    pub just_released: HashSet<T>,
```
Comment (memoryruins) [inline crates/bevy_input/src/input.rs:8]: #428 (not yet merged) adds methods that return iterators to these sets, which should remove the need for making these fields pub.

Comment (TheRawMeatball) [reply crates/bevy_input/src/input.rs:8]: This does look like a better idea - is there a way to merge pull requests, or should I just copy-paste the changes from there and do things that way?

Comment (memoryruins) [reply crates/bevy_input/src/input.rs:8]: It's probably fine to leave this as is for now until cart is able to give his feedback/direction.
First we should see if he thinks iterator returning methods are the correct approach anyway :)

Comment (stefee) [reply crates/bevy_input/src/input.rs:8]: You can't merge PRs exactly, but you can wait for one of the PRs to be merged into the main branch and then rebase your branch onto that change (i.e. rebase onto the latest main branch commits). You can technically rebase your branch onto the other branch in the meantime and set your PR's base branch to be the other branch, but this is probably confusing, and honestly, PRs that branch off of other PRs tend to become unwieldy quite fast.

Comment (TheRawMeatball) [reply crates/bevy_input/src/input.rs:8]: Ok, I'll wait thn

**Comment by TheRawMeatball** [comment]
Closing because of Axis API

**Pull request closed without merging**